src/libraries/Microsoft.Extensions.Hosting/src/Host.cs

```diff
@@ -55,5 +55,46 @@ public static IHostBuilder CreateDefaultBuilder(string[] args)
             HostBuilder builder = new();
             return builder.ConfigureDefaults(args);
         }
+
+#nullable enable
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HostApplicationBuilder"/> class with pre-configured defaults.
+        /// </summary>
+        /// <remarks>
+        ///   The following defaults are applied to the returned <see cref="HostApplicationBuilder"/>:
+        ///   <list type="bullet">
+        ///     <item><description>set the <see cref="IHostEnvironment.ContentRootPath"/> to the result of <see cref="Directory.GetCurrentDirectory()"/></description></item>
+        ///     <item><description>load host <see cref="IConfiguration"/> from "DOTNET_" prefixed environment variables</description></item>
+        ///     <item><description>load host <see cref="IConfiguration"/> from supplied command line args</description></item>
+        ///     <item><description>load app <see cref="IConfiguration"/> from 'appsettings.json' and 'appsettings.[<see cref="IHostEnvironment.EnvironmentName"/>].json'</description></item>
+        ///     <item><description>load app <see cref="IConfiguration"/> from User Secrets when <see cref="IHostEnvironment.EnvironmentName"/> is 'Development' using the entry assembly</description></item>
+        ///     <item><description>load app <see cref="IConfiguration"/> from environment variables</description></item>
+        ///     <item><description>load app <see cref="IConfiguration"/> from supplied command line args</description></item>
+        ///     <item><description>configure the <see cref="ILoggerFactory"/> to log to the console, debug, and event source output</description></item>
+        ///     <item><description>enables scope validation on the dependency injection container when <see cref="IHostEnvironment.EnvironmentName"/> is 'Development'</description></item>
+        ///   </list>
+        /// </remarks>
+        public static HostApplicationBuilder CreateApplicationBuilder() => new HostApplicationBuilder();
```

Comment:
How valuable are these methods? It feels wrong to have 2 public APIs that do exactly the same thing.

On the existing factory methods, We `ConfigureDefaults` over top of a `new HostBuilder()`. So for those it makes sense, since the ctor creates a blank builder, and the `CreateDefaultBuilder` creates one with certain things set.

But here, these factory APIs literally just call the ctor. So do we need both?

Reply:
During API review, I suggested adding just these static methods for discoverability. I agreed to add the equivalent constructors, but I think those are less important.